packages/device_info_plus/device_info_plus/ios/device_info_plus/Sources/device_info_plus/FPPDeviceInfoPlusPlugin.m

```diff
@@ -29,8 +30,10 @@ - (void)handleMethodCall:(FlutterMethodCall *)call
       isiOSAppOnMac = [NSNumber numberWithBool:[info isiOSAppOnMac]];
     }
     NSString *machine;
+    NSString *deviceName;
     if ([self isDevicePhysical]) {
       machine = @(un.machine);
+      deviceName = [DeviceIdentifiers userKnownDeviceModel:@(un.machine)];
     } else {
       machine = [info environment][@"SIMULATOR_MODEL_IDENTIFIER"];
     }
```

Comment:
Thanks for the changes! Could it be that `deviceName` remains null in the `else` case? later `modelName` is declared as a non-null String

Reply:
Good catch. Seems like this is the issue with integration tests. 
Updated to return model of the simulator.